image_pyramid.py

```diff
@@ -4,11 +4,17 @@
 class ImagePyramid:
     _ZOOMED_IN_LEVELS = 3  # Number of times you can zoom in beyond 100%
 
-    def __init__(self, matrix, sidelength):
+    def __init__(self, matrix, sidelength, hues=None):
```

Comment:
Off-topic: put in a docstring explaining that sidelength is the size of images we're going to return later.